jobserver/jobs/hourly/dump_db.py

```diff
@@ -30,6 +29,12 @@ def execute(self):
                 "--no-acl",
                 "--no-owner",
                 f"--file={output}",
-                database_url,
             ],
+            env={
+                "PGUSER": env.str("DATABASE_USER"),
```

Comment:
Possibly only the password needs to be in the environment? Thats how I've [done this before](https://github.com/canonical-ols/talisker/blob/master/talisker/postgresql.py#L72) - its actually useful to have the rest of connection string visible when we error.

I think you can then do, using urlparse as above, something like:

```
from urlib.parse import urlparse

db = urlparse(config.DATABASE_URL)

subprocess.check_output(
  [
    "pg_dump"
    f"postgresql://{db.user}@{db.netloc}{db.path}"
    ...
  ],
  env={"PGPASS": db.password},
)
```

Reply:
Thanks, that's a much nicer solution! I didn't like what I'd done, but it was late so I was hoping that something better would occur today. I didn't realise that urlparse would work, I was thinking it would have to be a regexp, which I wanted to avoid. I'll give that solution a try..